add mermaid flow diagram for patch releases (#8220)
cri-o/cri-o: scripts/automated-patch-releases.md

> # CRI-O Automated Patch Releases

Reviewer: Please include the mermaid diagrams directly here: https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/

Author: this link is not resolving `https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/`

Or do you mean I should copy the url of the automated-patch-releases.mermaid file and add it into this markdown file?

Reviewer: The link outlines only how to include mermaid in GitHub markdown. We do not need the `.mermaid` files, we can just include everything in one markdown file.

Author: I think it looks good now

# CRI-O Automated Patch Releases

> Automated Patch Releases provides an easy way of packaging and release new version

Reviewer: ```suggestion
Automated Patch Releases provide an easy way of packaging and releasing a new version
```

> of CRI-O
> This involves the use of Golang Scripts and Github Actions

Reviewer: ```suggestion
of CRI-O. This involves the use of Golang scripts and Github Actions and follows the
outlined flow:
```

```mermaid
    flowchart TD

>       A[Start: patch-release.yml Workflow] --> B[Create version bump intent prs]

Reviewer: Let's link the workflow files everywhere:
```suggestion
      A[Start: <a href="https://github.com/cri-o/cri-o/actions/workflows/patch-release.yml">patch-release.yml</a> Workflow] --> B[Create version bump intent prs]
```

>       A[Start: patch-release.yml Workflow] --> B[Create version bump intent prs]

Reviewer: ```suggestion
      A[Start: patch-release.yml Workflow] --> B[Create version bump PRs]
```

      B --> C[Maintainers Review PRs]

>       C -->|Decides Against Merges| E[Rebases on main branch] --> C

Reviewer: Mention the workflow as well as when (on the 1st of a month) the rebase happens.

      C -->|Decides to Merge PRs| D[Run: tag-reconciler.yml Workflow]
      D --> F[Release Job: Build Packages, Create Release Notes, etc.]

>       F --> G[End]

Reviewer: That's not the end. This reconcile job runs every day to create and publish the packages: https://github.com/cri-o/packaging/actions/workflows/schedule.yml

Author: What triggers the reconcile job, is it the <a href="https://github.com/cri-o/cri-o/actions/workflows/test.yml">test.yml</a> or is there something else that emits an event for that to run?

Reviewer: A cronjob: https://github.com/cri-o/packaging/blob/2b2aeaeaad98f87239e34f850e73b5957a41a111/.github/workflows/schedule.yml#L5

Author: okay but I am struggling to see how the workflows here, patch-release or  the test.yml which is last bit to run, relates to this schedule.yml. It looks like they are independent of each other. 
Also the schedule.yml is in a different repo so It's hard for me to visualize the relationship.

```

## Patch Release

> The `.github/workflows/patch-release.yml` runs the golang script

Reviewer: ```suggestion
The [`.github/workflows/patch-release.yml`](.github/workflows/patch-release.yml) runs the golang script
```

> `scripts/tag-reconciler/tag-reconciler.go`

Reviewer: Link the script here as well.

```mermaid
    flowchart TD
        A[Start patch-release Github action] --> B[Run Release Script]
        --> C[Get latest Minor versions from release Branches]
        C --> D[Bump up version in specific release branch]
           D -->E[Create PR with new version]
        E--> F[End]
```

## Pushing New Version tags

To push the new version tags and cut the release, the
`.github/workflows/tag-reconciler.yml` runs the golang script
`scripts/tag-reconciler/tag-reconciler.go`

This will inturn run the `.github/workflows/test.yml` which will build and

> package the latest versions for CRI-O

Reviewer: Please link those files

```mermaid
    flowchart TD

>         A[Start tag-reconciler Github action] --> B[Run Tag-Reconciler Script]

Reviewer: ```suggestion
        A[Start tag-reconciler Github action] --> B[Run tag-reconciler.go Script]
```

>         --> C[Check for latest minor versions Tags on remote]

Reviewer: ```suggestion
        --> C[Check for latest minor versions tags on remote]
```
